src/main/java/org/opentripplanner/middleware/controllers/api/ApiKeyManagementController.java

```diff
@@ -0,0 +1,150 @@
+     */
+    public static void register(Service spark, String apiPrefix) {
+        // FIXME need to agree on endpoint (/admin or /secure or a mix) and endpoint names
+        spark.get(apiPrefix + "/admin/apikey/create", ApiKeyManagementController::createApiKeyForApiUser, JsonUtils::toJson);
+        spark.get(apiPrefix + "/admin/apikey/delete", ApiKeyManagementController::deleteApiKeyForApiUser, JsonUtils::toJson);
+        spark.get(apiPrefix + "/admin/apikey/usage", ApiKeyManagementController::getUsageLogsForApiUser, JsonUtils::toJson);
+    }
+
+    /**
+     * Create a new API key and assign it to the provided usage plan. If no usage plan is provided use the default
+     * usage plan instead.
+     */
+    private static CreateApiKeyResult createApiKeyForApiUser(Request req, Response res) {
+
+        ApiUser user = getApiUser(req);
+
+        String usagePlanId = req.queryParamOrDefault("usagePlanId", DEFAULT_USAGE_PLAN_ID);
+        // FIXME Should an Api user be limited to one api key per usage plan (and perhaps stage)?
+        CreateApiKeyResult apiKey = ApiGatewayUtils.createApiKey(user.id, usagePlanId);
+        if (apiKey == null) {
+            logMessageAndHalt(req,
+                HttpStatus.INTERNAL_SERVER_ERROR_500,
+                String.format("Unable to get AWS api key for user (%s)", user),
+                null);
+        }
+
+        // add new api key to user and persist
+        user.apiKeyIds.add(apiKey.getId());
+        apiUsers.replace(user.id, user);
+
+        // FIXME assumption is that the api key will be required by the calling service
+        return apiKey;
+    }
+
+    /**
+     * Delete an api key from a given user's list of api keys (if present) and from AWS api gateway.
+     */
+    private static boolean deleteApiKeyForApiUser(Request req, Response res) {
+
+        ApiUser user = getApiUser(req);
+
+        String apiKeyId = req.queryParamOrDefault("apiKeyId", null);
+        if (apiKeyId == null) {
+            logMessageAndHalt(req,
+                HttpStatus.BAD_REQUEST_400,
+                "An api key id is required",
+                null);
+        }
+
+        // FIXME perhaps not required?
+        if (!user.apiKeyIds.contains(apiKeyId)) {
+            logMessageAndHalt(req,
+                HttpStatus.BAD_REQUEST_400,
+                String.format("User id (%s) does not have expected api key id (%s)", user.id, apiKeyId),
+                null);
+        }
+
+        // delete api key from user and persist
+        user.apiKeyIds.remove(apiKeyId);
+        apiUsers.replace(user.id, user);
+
+        // delete api key from AWS
+        ApiGatewayUtils.deleteApiKeys(Collections.singletonList(apiKeyId));
+
+        return true;
+    }
+
+    /**
+     * Get API usage from AWS api gateway for a users api keys.
+     */
+    private static List<GetUsageResult> getUsageLogsForApiUser(Request req, Response res) {
```

Comment:
I was thinking that this endpoint would just go under the `LogController`.

Reply:
Updated.